### PR TITLE
feature-health-check

### PR DESCRIPTION
**Pull request by shahn95**
Adding health-check functionality, work-in-progress.

### CODE REVIEW

File: src/chaat/app.clj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+(defn start-server
+  [val]
+  (jetty/run-jetty app {:port 3000
+                        :join? false}))
+
+(defn -main
+  [& args]
+  (swap! server start-server))
+
+;; (.stop server)
```
Comment (neenaoffline) [inline src/chaat/app.clj]: Prefer `(comment ...)` blocks for structural editing support

File: src/chaat/handler.clj [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  (:require [ring.util.response :as res]
+            [clj-time.local :as l]))
+
+(defn home
+  [request]
+  (res/response "Welcome to chaat"))
+
+(defn health-check
+  [request]
+  (->> (l/local-now)
```
Comment (neenaoffline) [inline src/chaat/handler.clj]: Prefer using threading to describe a sequence of operations to a singular data structure.

```clojure

(->> cart
 (add-items items)
 (apply-discount-code code)
 (apply-shipping-price address)
 (apply-taxes pincode))
```

I wouldn't use threading here for this.

File: test/chaat/core_test.clj [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
+
 (ns chaat.core-test
   (:require [clojure.test :refer :all]
-            [chaat.core :refer :all]))
+            [chaat.handler :as handler]
+            [clj-time.local :as l]))
+
+(defn time-stub
+  []
+  "mock-time")
```
Comment (neenaoffline) [inline test/chaat/core_test.clj]: Use a realistic stub when possible.

**Review by neenaoffline — APPROVED** [review]

File: src/chaat/app.clj (hunk 1 above)
Comment (shahn95) [reply src/chaat/app.clj]: Yes, done.

File: src/chaat/handler.clj (hunk 2 above)
Comment (shahn95) [reply src/chaat/handler.clj]: Hmm, this makes sense. I've modified this to describe a sequence (only one here) of operations to a singular data structure.

File: test/chaat/core_test.clj (hunk 3 above)
Comment (shahn95) [reply test/chaat/core_test.clj]: Yes, done.

File: src/chaat/handler.clj (hunk 2 above)
Comment (shahn95) [reply src/chaat/handler.clj]: Rebased a previous commit to include this refactor.

Comment (shahn95) [reply src/chaat/handler.clj]: You mentioned you wouldn't use threading here. What do you think of my final modification?
Would a `let` block be better, or would you just use plain function composition?
I'm thinking of this from a readability point of view.

**Pull request merged**